### ISSUE TITLE
space-infix-ops: option to ignore space between `=` and arrow function params?

### ISSUE THREAD

**Issue by aubergine10** [eslint/eslint#6428]
This is personal preference thing, but thought I'd mention it in hope that it's something which could be implemented without too much pain...

Rule: http://eslint.org/docs/rules/space-infix-ops

**What version of ESLint are you using?**

2.11.1

**What parser (default, Babel-ESLint, etc.) are you using?**

Default (as far as I can tell, I'm using SublimeLinter package for SublimeText 3).

**Please show your full configuration:**

```javascript
module.exports = {
  "env": {
    "es6": true,
    "node": true
  },

  "extends": "eslint:recommended",

  "rules": {
    "space-infix-ops": [
      "error",
      {
        "int32Hint": false
        // would be nice to have new option here for ignoring
        // space between `=` and arrow function params
      }
    ],

    "template-curly-spacing": [
      "warn",
      "never"
    ],
  }
};
```

**What did you do? Please include the actual source code causing the issue.**

```javascript
let foo =a=> { // JS equivalent of "go faster" stripes, oddly satisfying
  //    ^ allow no space between `=` and params
}

// also

let bar =(a,b)=> {
  //    ^ allow no space between `=` and params
}
```

**What did you \[want] to happen?**

It would be nice to have a rule option to ignore spacing between `=` and arrow function params (similar approach to the [`int32Hint` option](http://eslint.org/docs/rules/space-infix-ops#options)), because arrow functions look even better when they are like javelins skewering the function parameters FTW.

**What actually happened? Please include the actual, raw output from ESLint.**

It flagged the `=` as not having a space between it and the parameters of the arrow function.

I'm not sure how to get the actual raw output from ESLint from within Sublime Text :s

**Comment by ilyavolodin** [comment]
@eslint/eslint-team Thoughts? I don't use arrow function, so I have no preferences for style. However, this seems like a very personal style requirement. Not sure it fits the core.

**Comment by mysticatea** [comment]
I felt this is a very personal style, too.
I think this is proper to a plugin.

**Comment by aubergine10** [comment]
I'm looking at the rule source to see if I can work out how to adapt it to handle arrow functions differently, but the "Technical Deficit" indicator has lit up in my head and is currently flashing feverishly. Any pointers as to what I need to read to get the pre-requisite knowledge of how to adapt the vanilla rule to my requirements?

My current assumption is that I need to modify the [`checkVar()`](https://github.com/eslint/eslint/blob/master/lib/rules/space-infix-ops.js#L141) method to identify arrow function expressions and treat them differently?
